docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,491 @@
+
+    - ![Manual dispatch](./media/actions/deploy/manual-dispatch.jpg)
+    **Figure 13**: Manual dispatch.
+
+## Handle Environment Configuration
+
+Your workflow is deploying the same binary to each environment. This is an important concept to ensure that the binaries you test in one environment are the same that you deploy to the next. However, environments typically have different settings like database connection strings: you want to ensure that the `DEV` application is using `DEV` settings and the `PROD` application is using `PROD` settings!
+
+For this simple application there is no database connection string. However, there is an example configuration setting that you can modify for each environment. If you open the `simple-feed-reader/SimpleFeedReader/appsettings.json` file you'll see that the configuration includes a setting for the Header text on the Index page:
+
+```yml
+  "UI": {
+    "Index": {
+      "Header": "Simple News Reader"
+    }
+  },
+```
+
+To show how environment configuration can be handled, you're going to add a secret to each environment and then substitute that value into the settings as you deploy.
+
+### Add Environment Secrets
+
+1. Select `Settings` on your repository and then `Environments` and select `PRE-PROD`.
+1. Select `Add secret` and add a secret called `index_header` with the value `PRE PROD News Reader`. Select `Add secret`.
+
+    - ![Add an environment secret](./media/actions/deploy/add-env-secret.jpg)
+    **Figure 14**: Add an environment secret.
+
+1. Repeat these steps to add a secret called `index_header` with the value `PROD News Reader` for the `PROD` environment.
+1. If you select `Settings` and `Secrets` in the repository, you'll see the changes. They should look something like this:
+
+    - ![View secrets](./media/actions/deploy/env-secrets.jpg)
+    **Figure 15**: View secrets.
+
+### Update the workflow to handle configuration
+
+1. Navigate to the `.github/workflows/dotnet.yml` file and select the pencil icon to edit the file.
+1. Add the following step before the `az cli logout` step in the `deploy_staging` job:
+
+    ```yml
+        - name: Update config
+          uses: Azure/appservice-settings@v1
+          with:
+          app-name: ${{ env.app-name }}
+          slot-name: staging
+          app-settings-json: |
+            [
+                {
+                    "name": "UI:Index:Header",
+                    "value": "${{ secrets.INDEX_HEADER }}",
+                    "slotSetting": true
+                }
+            ]
+        
+        - name: az cli logout   # <-- this exists already
+    ```
+
+1. Add almost the same code to the `deploy_prod` job above its `az cli logout` step. The only difference is that you do not specify a `slot-name`, since you're targeting the production slot:
```

Comment:
```suggestion
1. Add almost the same code to the `deploy_prod` job above its `az cli logout` step. The only difference is that you don't specify a `slot-name`, since you're targeting the production slot:
```